pkgs/by-name/po/porn-vault/package.nix

```diff
@@ -0,0 +1,112 @@
+{
+  fetchFromGitLab,
+  fetchurl,
+  lib,
+  pnpm,
+  stdenvNoCC,
+  nodejs_22,
+  ffmpeg,
+  imagemagick,
+  writeShellScript,
+  makeWrapper,
+  autoPatchelfHook,
+}:
+let
+  izzy = stdenvNoCC.mkDerivation rec {
+    pname = "izzy";
+    version = "2.0.2";
+
+    src = fetchurl {
+      url = "https://gitlab.com/porn-vault/izzy/-/package_files/151366122/download";
+      hash = "sha256-K+rVb2F3HizMKIpY1zbLj5zE7B6DkIoq8nJYPUpQn+Y=";
+    };
+
+    dontUnpack = true;
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+    ];
+
+    installPhase = ''
+      install -Dm755 $src $out/bin/izzy
+    '';
+
+    meta = with lib; {
+      description = "Rust In-Memory K-V Store with Redis-Style File Persistence and Secondary Indices";
+      homepage = "https://gitlab.com/porn-vault/izzy";
+      license = lib.licenses.gpl3;
+      maintainers = with maintainers; [ luNeder ];
+      platforms = [ "x86_64-linux" ];
+      mainProgram = "izzy";
+    };
+  };
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "porn-vault";
+  version = "0.30.0-rc.11";
+
+  src = fetchFromGitLab {
+    owner = "porn-vault";
+    repo = "porn-vault";
+    rev = "4c6182c5825d85193cf67cb7cd927da2feaaecdb";
+    hash = "sha256-wQ3dqLc0l2BmLGDYrbWxX2mPwO/Tqz0fY/fOQTEUv24=";
+  };
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-Xr9tRiP1hW+aFs9FnPvPkeJ0/LtJI57cjWY5bZQaRTQ=";
+  };
+
+  nativeBuildInputs = [
+    nodejs_22
+    pnpm.configHook
+    makeWrapper
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm build
+    sed s+%out%+$out+ ${./porn-vault.sh} > ./porn-vault.sh
+    sed s+%nodejs%+${nodejs_22}+ -i ./porn-vault.sh
+    sed s+%izzy%+${izzy}+ -i ./porn-vault.sh
```

Comment:
Why not `substituteInPlace`?

Reply:
Bc I didn’t know that existed lmao

Is that preferred over sed?

Reply:
Yup

Reply:
I think [substituteAll](https://nixos.org/manual/nixpkgs/unstable/#fun-substituteAll) would be a better choice here, as it's what LuNeder has basically re-implemented. Could be easily adapted to this

```
nodejs=${nodejs}
izzy=${izzy}
substituteAll ${./porn-vault.sh} "$out"/bin/porn-valut
```

(note: `out` is not set here as it's already an env var)